classes/Http/Controller/TalkFormController.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2017 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Http\Controller;
+
+use OpenCFP\Domain\CallForPapers;
+use OpenCFP\Domain\Model\Talk;
+use OpenCFP\Domain\Services\Authentication;
+use OpenCFP\Http\View\TalkHelper;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Twig_Environment;
+
+class TalkFormController extends BaseController
+{
+    private $authentication;
+    private $talkHelper;
+    private $callForPapers;
+
+    public function __construct(
+        Authentication $authentication,
+        TalkHelper $talkHelper,
+        CallForPapers $callForPapers,
+        Twig_Environment $twig,
+        UrlGeneratorInterface $urlGenerator
+    ) {
+        $this->authentication = $authentication;
+        $this->talkHelper     = $talkHelper;
+        $this->callForPapers  = $callForPapers;
+
+        parent::__construct($twig, $urlGenerator);
+    }
+
+    public function editAction(Request $request)
+    {
+        $talkId = (int) $request->get('id');
+        // You can only edit talks while the CfP is open
+        // This will redirect to "view" the talk in a read-only template
+        if (!$this->callForPapers->isOpen()) {
+            $request->getSession()->set(
+                'flash',
+                [
+                    'type'  => 'error',
+                    'short' => 'Read Only',
+                    'ext'   => 'You cannot edit talks once the call for papers has ended', ]
```

Comment:
What do you think about adjusting the wrapping here to

```diff
diff --git a/classes/Http/Controller/TalkFormController.php b/classes/Http/Controller/TalkFormController.php
index bb4cba0d..51b44ac9 100644
--- a/classes/Http/Controller/TalkFormController.php
+++ b/classes/Http/Controller/TalkFormController.php
@@ -54,7 +54,8 @@ class TalkFormController extends BaseController
                 [
                     'type'  => 'error',
                     'short' => 'Read Only',
-                    'ext'   => 'You cannot edit talks once the call for papers has ended', ]
+                    'ext'   => 'You cannot edit talks once the call for papers has ended',
+                ]
             );

             return new RedirectResponse($this->url('talk_view', ['id' => $talkId]));
```

or

```diff
diff --git a/classes/Http/Controller/TalkFormController.php b/classes/Http/Controller/TalkFormController.php
index bb4cba0d..ad88ffed 100644
--- a/classes/Http/Controller/TalkFormController.php
+++ b/classes/Http/Controller/TalkFormController.php
@@ -49,13 +49,11 @@ class TalkFormController extends BaseController
         // You can only edit talks while the CfP is open
         // This will redirect to "view" the talk in a read-only template
         if (!$this->callForPapers->isOpen()) {
-            $request->getSession()->set(
-                'flash',
-                [
-                    'type'  => 'error',
-                    'short' => 'Read Only',
-                    'ext'   => 'You cannot edit talks once the call for papers has ended', ]
-            );
+            $request->getSession()->set('flash', [
+                'type'  => 'error',
+                'short' => 'Read Only',
+                'ext'   => 'You cannot edit talks once the call for papers has ended',
+            ]);

             return new RedirectResponse($this->url('talk_view', ['id' => $talkId]));
         }
```